packages/zero-tag-processor/src/styled.ts

```diff
@@ -0,0 +1,452 @@
+import { BaseProcessor, buildSlug, toValidCSSIdentifier, validateParams } from '@linaria/tags';
+import type {
+  Expression,
+  Params,
+  TailProcessorParams,
+  ValueCache,
+  IOptions as IBaseOptions,
+  WrappedNode,
+} from '@linaria/tags';
+import { ValueType, slugify } from '@linaria/utils';
+import type {
+  IVariableContext,
+  Rules,
+  Replacements,
+  ExpressionValue,
+  LazyValue,
+} from '@linaria/utils';
+import type { Theme } from '@mui/material/styles';
+import type { ParseResult } from '@babel/parser';
+import { parseExpression } from '@babel/parser';
+import type { SourceLocation } from '@babel/types';
+import cssFnValueToVariable from './utils/cssFnValueToVariable';
+import processCssObject from './utils/processCssObject';
+import valueToLiteral from './utils/valueToLiteral';
+
+type VariantData = {
+  props: Record<string, string | number | boolean | null>;
+  style: object;
+};
+
+type VariantDataTransformed = {
+  props: Record<string, string | number | boolean | null>;
+  className: string;
+};
+
+export type CustomOptions = {
+  themeArgs?: {
+    theme: Theme;
+  };
+};
+
+type IOptions = IBaseOptions & CustomOptions;
+type ComponentNames = keyof Exclude<Theme['components'], undefined>;
+
+type ComponentMeta = {
+  name?: ComponentNames;
+  // slot?: string;
+  skipVariantsResolver?: boolean;
+  skipSx?: boolean;
+};
+
+/**
+ * Linaria tag processor responsible for converting complex `styled()()` calls
```

Comment:
This is a great documentation for the method. Can we add similar description on the other functions defined too (for both packages)? It would help with the reviews a lot.